includes/documents/abstract-wcpdf-order-document.php

```diff
@@ -1364,6 +1364,38 @@ public function get_number_store_year( $table_name ) {
 		return intval( $year );
 	}
 
+	/**
+	 * Returns the due date timestamp.
+	 *
+	 * @return int
+	 */
+	public function get_due_date(): int {
+		if ( empty( $this->order ) || empty( $this->settings['due_date'] ) ) {
+			return 0;
+		}
+
+		$due_date_days = ( 'custom' === $this->settings['due_date'] ) ? intval( $this->settings['due_date_custom_days'] ?? 0 ) : absint( $this->settings['due_date'] );
+		$due_date_days = apply_filters( "wpo_wcpdf_{$this->slug}_due_date_days", $due_date_days, $this );
+
+		if ( 0 >= $due_date_days ) {
+			return 0;
+		}
+
+		if ( isset( $this->settings['due_date_base_date'] ) && 'invoice_date' === $this->settings['due_date_base_date'] && $this->exists() ) {
```

Comment:
Maybe we could use the document type here? `"{$this->get-type()}_date"`

Reply:
Ah, missed that.
I think changing the key to "document_date" in settings would be better.
What do you think?

Reply:
Yes, I agree.